packages/playground-examples/copy/ja/TypeScript/Meta-Types/Indexed Types.ts

```diff
@@ -0,0 +1,37 @@
+// 重複した型に気付く場合があります。
+// よくある例は自動生成されたAPIレスポンスの
+// ネストされたリソースです。
+
+interface ArtworkSearchResponse {
+  artists: {
+    name: string;
+    artworks: {
+      name: string;
+      deathdate: string | null;
+      bio: string;
+    }[];
+  }[];
+}
+
+// もし、このinterfaceが手動で作られていたら、
+// 以下のようにartworksからinterfaceを抜き出してくることはとても簡単です。
+
+interface Artwork {
+  name: string;
+  deathdate: string | null;
+  bio: string;
+}
+
+// しかし、この場合、APIをいじることはできません。
+// 手動でinterfaceをArtworkSearchResponseの
+// 一部であるartworksとして定義することも可能ですが、
+// するとArtwork型はレスポンスが変更されたときに同期されません。
+
+// これに対する工夫はJavaScriptが
+// 文字列でプロパティにアクセスする際に用いる手法の複製であるindexed typeです
```

Comment:
[IMO]

以下のように、訳してみましたがいかがでしょうか？

```suggestion
// これに対する修正方法は、indexed typeです。
// indexed typeは、JavaScriptの文字列を使ったプロパティへのアクセス方法を複製しています。
```

(複製より、模倣とかのほうが良さそうな気がしますが :thinking: )

Reply:
模倣とかの方が通りが良さそうですね
```suggestion
// これに対する修正方法は、indexed typeです。
// indexed typeは、JavaScriptの文字列を使ったプロパティへのアクセス方法を模倣しています。
```